### PR TITLE
Update docs for make/config target changes

### PR DESCRIPTION
**Pull request by hamishwillee**
Update for changes in https://github.com/PX4/Firmware/pull/10875

### CODE REVIEW

File: en/hardware/porting_guide.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ This guide is focused only on the host OS and middleware as the applications/fli
 In addition to the host operating system specific configuration files described below, there are several groups of configuration files for each board located throughout the code base:
```
Comment (hamishwillee) [inline en/hardware/porting_guide.md]: @dagar Can you please let me know correct paths for everything in this section (porting guide)

File: en/debug/simulation_debugging.md [hunk 2]
```diff
@@ -30,21 +30,21 @@ sudo apt-get install valgrind
...
+make px4_sitl_default jmavsim___lldb
 
-make posix_sitl_default gazebo
-make posix_sitl_default gazebo___gdb
-make posix_sitl_default gazebo___lldb
+make px4_sitl_default gazebo
+make px4_sitl_default gazebo___gdb
+make px4_sitl_default gazebo___lldb
 
 make posix_sitl_lpe jmavsim
```
Comment (hamishwillee) [inline en/debug/simulation_debugging.md]: Note to me, all these can be removed.

**Comment by hamishwillee** [comment]
**uorb_graph.md** refers to `<option value='graph_px4fmu-v4.json'>FMUv4 Modules</option> ` - will that path for the json change?

File: kr/debug/gdb_debugging.md [hunk 3]
```diff
@@ -128,7 +126,7 @@ EXC_RETURN: ffffffe9
 <div class="host-code"></div>
 
 ```bash
-arm-none-eabi-gdb build/px4fmu-v2_default/src/firmware/nuttx/firmware_nuttx
+arm-none-eabi-gdb build/px4_fmu-v2_default/src/firmware/nuttx/firmware_nuttx
```
Comment (hamishwillee) [inline kr/debug/gdb_debugging.md:129]: Is the rest of the path in this still oK? ... /src/firmware/nuttx/firmware_nuttx

File: kr/debug/profiling.md [hunk 4]
```diff
@@ -33,13 +33,13 @@ PMSP는 GDB를 사용하여 stack trace를 수집합니다.
 사용 예제:
 
 ```bash
-./poor-mans-profiler.sh --elf=build/px4fmu-v4_default/src/firmware/nuttx/firmware_nuttx --nsamples=30000
+./poor-mans-profiler.sh --elf=build/px4_fmu-v4_default/src/firmware/nuttx/firmware_nuttx --nsamples=30000
```
Comment (hamishwillee) [inline kr/debug/profiling.md]: ditto - is that full path after --elf= correct?

File: kr/debug/simulation_debugging.md [hunk 5]
```diff
@@ -30,13 +30,13 @@ sudo apt-get install valgrind
...
+make px4_sitl_default jmavsim___lldb
 
-make posix_sitl_default gazebo
-make posix_sitl_default gazebo___gdb
-make posix_sitl_default gazebo___lldb
+make px4_sitl_default gazebo
+make px4_sitl_default gazebo___gdb
+make px4_sitl_default gazebo___lldb
 
 make posix_sitl_lpe jmavsim
```
Comment (hamishwillee) [inline kr/debug/simulation_debugging.md]: Note to me, remove these.

File: kr/debug/profiling.md (hunk 4 above)
Comment (dagar) [reply kr/debug/profiling.md]: No, that's been wrong for a long time.

Comment (dagar) [reply kr/debug/profiling.md]: ./build/px4_fmu-v4_default/px4_fmu-v4_default.elf

File: kr/debug/gdb_debugging.md (hunk 3 above)
Comment (dagar) [reply kr/debug/gdb_debugging.md:129]: ./build/px4_fmu-v2_default/px4_fmu-v2_default.elf

File: en/hardware/porting_guide.md [hunk 6]
```diff
@@ -13,7 +13,7 @@ This guide is focused only on the host OS and middleware as the applications/fli
...
 * Board startup and configuration files are located in: [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards).
   * This folder contains bus mappings, GPIO mappings, and the initialization code for each board.
-  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5).
+  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: https://github.com/PX4/Firmware/tree/master/boards/px4/fmu-v5/src

File: en/hardware/porting_guide.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ This guide is focused only on the host OS and middleware as the applications/fli
 In addition to the host operating system specific configuration files described below, there are several groups of configuration files for each board located throughout the code base:
 * Board startup and configuration files are located in: [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: Are located in each board's directory (board/VENDOR/MODEL/)

Comment (dagar) [reply en/hardware/porting_guide.md]: https://github.com/PX4/Firmware/tree/master/boards/

File: en/hardware/porting_guide.md [hunk 8]
```diff
@@ -13,7 +13,7 @@ This guide is focused only on the host OS and middleware as the applications/fli
...
   * This folder contains bus mappings, GPIO mappings, and the initialization code for each board.
-  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5).
+  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5). 
 * The boot file system (startup script) is located in: [ROMFS/px4fmu\_common](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common)
 * The board specific build configurations are located in: [cmake/configs/](https://github.com/PX4/Firmware/blob/master/cmake/configs/).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: This is now in the same location (board/VENDOR/MODEL/).

File: en/hardware/porting_guide.md [hunk 9]
```diff
@@ -31,20 +31,20 @@ For all NuttX based flight controllers (e.g. the Pixhawk series) the OS is loade
 The configuration files for NuttX based boards, including linker scripts and other required settings are located under [platforms/nuttx/nuttx-configs](https://github.com/PX4/Firmware/tree/master/platforms/nuttx/nuttx-configs).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: Now in the same location (board/VENDOR/MODEL/nuttx-config)

File: en/hardware/porting_guide.md [hunk 10]
```diff
@@ -31,20 +31,20 @@ For all NuttX based flight controllers (e.g. the Pixhawk series) the OS is loade
...
 
 The following example uses FMUv5 as it is a recent [reference configuration](../debug/reference-design.md) for NuttX based flight controllers:
-* Running `make px4fmu-v5_default` from the `src/Firmware` directory will build the FMUv5 config
+* Running `make px4_fmu-v5_default` from the **Firmware** directory will build the FMUv5 config
 * The base FMUv5 configuration files are located in: [platforms/nuttx/nuttx-configs/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/platforms/nuttx/nuttx-configs/px4fmu-v5).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: Same location - eg fmu-v5 NuttX config - https://github.com/PX4/Firmware/tree/master/boards/px4/fmu-v5/nuttx-config

File: en/hardware/porting_guide.md [hunk 11]
```diff
@@ -31,20 +31,20 @@ For all NuttX based flight controllers (e.g. the Pixhawk series) the OS is loade
...
 * Board specific header: [platforms/nuttx/nuttx-configs/px4fmu-v5/include/board.h](https://github.com/PX4/Firmware/blob/master/platforms/nuttx/nuttx-configs/px4fmu-v5/include/board.h).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: https://github.com/PX4/Firmware/blob/master/boards/px4/fmu-v5/nuttx-config/include/board.h

File: en/hardware/porting_guide.md [hunk 12]
```diff
@@ -31,20 +31,20 @@ For all NuttX based flight controllers (e.g. the Pixhawk series) the OS is loade
...
 * NuttX OS config (created with Nuttx menuconfig): [nuttx-configs/px4fmu-v5/nsh/defconfig](https://github.com/PX4/Firmware/blob/master/platforms/nuttx/nuttx-configs/px4fmu-v5/nsh/defconfig).
```
Comment (dagar) [inline en/hardware/porting_guide.md]: https://github.com/PX4/Firmware/blob/master/boards/px4/fmu-v5/nuttx-config/nsh/defconfig

**Comment by dagar** [comment]
> **uorb_graph.md** refers to `<option value='graph_px4fmu-v4.json'>FMUv4 Modules</option> ` - will that path for the json change?

Yes, although I don't see that being automatically uploaded anywhere. Let's review that separately.

File: en/hardware/porting_guide.md [hunk 13]
```diff
@@ -10,12 +10,14 @@ This guide is focused only on the host OS and middleware as the applications/fli
...
-  * This folder contains bus mappings, GPIO mappings, and the initialization code for each board.
-  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5).
+Board startup and configuration files are located under [/boards](https://github.com/PX4/Firmware/tree/master/boards/) in each board's vendor-specific directory (i.e. **boards/_VENDOR_/_MODEL_/**)).
+
+For example, for FMUv5:
+* (All) Board-specific files: [/boards/px4/fmu-v5](https://github.com/PX4/Firmware/tree/master/boards/px4/fmu-v5). 
+* Build configuration: [/boards/px4/fmu-v5/default.cmake](https://github.com/PX4/Firmware/blob/master/boards/px4/fmu-v5/default.cmake).
+  
+In addition there are several groups of configuration files for each board located throughout the code base:
 * The boot file system (startup script) is located in: [ROMFS/px4fmu\_common](https://github.com/PX4/Firmware/tree/master/ROMFS/px4fmu_common)
```
Comment (hamishwillee) [inline en/hardware/porting_guide.md:20]: @dagar Is this true?

**Comment by hamishwillee** [comment]
@dagar As discussed, I have merged the devguide changes and we can iterate.

HOWEVER, as a priority the porting guide needs your further review "as a whole" following this update. ie it was written assuming host OS was the main separator and config was scattered. That is no longer true and we need to a) capture that and b) make sure that other relevant information is available.

File: kr/debug/gdb_debugging.md (hunk 3 above)
Comment (hamishwillee) [reply kr/debug/gdb_debugging.md:129]: Thanks. Fixed in https://github.com/PX4/Devguide/commit/d0443c97fe204da65b9d4b34e54788f4aabaa8d0

**Pull request merged**